packages/preferences/src/browser/views/components/preference-select-input.ts

```diff
@@ -48,26 +79,25 @@ export class PreferenceSelectInputRenderer extends PreferenceLeafNodeRenderer<JS
     }
 
     protected doHandleValueChange(): void {
-        const currentValue = this.interactable.value || undefined;
         this.updateInspection();
         const newValue = this.getDataValue();
         this.updateModificationStatus(this.getValue());
-        if (newValue !== currentValue && document.activeElement !== this.interactable) {
-            this.interactable.value = newValue;
+        if (document.activeElement !== this.interactable) {
+            this.onDidChangeEmitter.fire(newValue);
         }
     }
 
     /**
      * Returns the stringified index corresponding to the currently selected value.
      */
-    protected getDataValue(): string {
+    protected getDataValue(): number {
         const currentValue = this.getValue();
         const selected = this.enumValues.findIndex(value => value === currentValue);
```

Comment:
To myself a few months ago: since enum values can be objects, this should probably be a deep equality check.